Lib/tarfile.py

```diff
@@ -214,7 +215,7 @@ def itn(n, digits=8, format=DEFAULT_FORMAT):
             s.insert(1, n & 0o377)
             n >>= 8
     else:
-        raise ValueError("overflow in number field")
+        raise ValueError("overflow in number field processing %r as %r" % (original_n, format))
```

Comment:
I do not think it is useful. It would be more helpful to report which kind of value (size, mtime, etc) is overflowed instead of just its numerical value without connection to the name, but this information is not available here. `format` is not informative either. It is an integer 0 to 2, and the user usually know what format is used.

Reply:
At some future date I hope to make `format` a symbolic constant similar to `Enum`.  I agree that "kind" would be more useful, but that would be a bigger change than I am ready to do at the moment.